File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/interfaceInJava/interfaceInheritance/Person.java
package de.fraunhofer.iem.classesAndObjects.interfaceInJava.interfaceInheritance;

/**
 * Implements the interface that has multiple inheritance
 *
 * @author Ranjith Krishnamurthy
 */
public class Person implements Movable {
    @Override
    public void jog() {
        System.out.println("Jogging");
    }

    @Override
    public void run() {
        System.out.println("Running");
    }

    @Override
    public void walk() {
        System.out.println("Walking");
    }

    @Override
    public void crawl() {
        System.out.println("Crawling");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/interfaceInKotlin/interfaceInheritance/Person.kt
package de.fraunhofer.iem.classesAndObjects.interfaceInKotlin.interfaceInheritance

/**
 * Implements the interface that has multiple inheritance
 *
 * @author Ranjith Krishnamurthy
 */
class Person : Movable {
    override fun jog() {
        println("Jogging")
    }

    override fun run() {
        println("Running")
    }

    override fun walk() {
        println("Walking")
    }

    override fun crawl() {
        println("Crawling")
    }
}